addons-cluster/apecloud-mysql-cluster/templates/etcd-service-descriptor.yaml

```diff
@@ -7,4 +8,9 @@ spec:
   serviceKind: etcd
   serviceVersion: {{ .Values.etcd.serviceVersion }}
   endpoint:
-    value: {{ .Values.etcd.serviceReference.endpoint }}
\ No newline at end of file
+    {{- if .Values.localEtcd }}
+    value: {{ include "kblib.clusterName" . }}-etcd-headless.default.svc.cluster.local:2379
+    {{- else }}
```

Comment:
The ServiceDescriptor represents an external etcd. If it's an etcd component within the cluster, use vars.componentVars.

Reply:
btw, here it seems you should not use headless svc.

Reply:
This part is saying that in order to standardize the use of serviceRef, I simply created a branch here. The value of the internal etcd should be fixed. 
Initially, I used Vars to reference componentVars. But it seemed a bit complicated.Should I change it back?

Reply:
> This part is saying that in order to standardize the use of serviceRef, I simply created a branch here. The value of the internal etcd should be fixed. Initially, I used Vars to reference componentVars. But it seemed a bit complicated.Should I change it back?

Yes, always use ServiceDescriptor to configure the address of external services. I'm discussing with @leon-inf to add support for dynamically choosing and representing both external services and component services based on an environment variable.

Reply:
for now, maybe you should define both EXTERNAL_ETCD_SERVICE_ENDPOINT and ETCD_COMP_SERVICE_ENDPOINT vars, and write a sh function getEndpoint() to resolve.